container-core/src/main/java/com/yahoo/restapi/RestApiImpl.java

```diff
@@ -151,11 +152,12 @@ private static Route createDefaultRoute() {
 
     private static List<ExceptionMapperHolder<?>> combineWithDefaultExceptionMappers(
             List<ExceptionMapperHolder<?>> configuredExceptionMappers, boolean disableDefaultMappers) {
-        List<ExceptionMapperHolder<?>> exceptionMappers = new ArrayList<>();
+        List<ExceptionMapperHolder<?>> exceptionMappers = new ArrayList<>(configuredExceptionMappers);
         if (!disableDefaultMappers){
             exceptionMappers.add(new ExceptionMapperHolder<>(RestApiException.class, (context, exception) -> exception.response()));
         }
-        exceptionMappers.addAll(configuredExceptionMappers);
+        // Topologically sort children before superclasses, so most the specific match is found by iterating through mappers in order.
+        exceptionMappers.sort((a, b) -> a.type.isAssignableFrom(b.type) ? 1 : b.type.isAssignableFrom(a.type) ? -1 : 0);
```

Comment:
Is comparator correct for case when a.type == b.type ?

Reply:
No, but perhaps that should be detected construction time as a user error? Or perhaps ee should Allie overprising the default ones, but only once?

Reply:
```suggestion
        exceptionMappers.sort((a, b) -> (a.type.isAssignableFrom(b.type) ? 1 : 0) + (b.type.isAssignableFrom(a.type) ? -1 : 0));
```